src/Makefile

```diff
@@ -53,6 +53,9 @@ ENABLE_DOUBLE_VLAN = n
 ENABLE_PAGE_REUSE = n
 ENABLE_RX_PACKET_FRAGMENT = n
 
+# GCC 14 will report the following warning as an error and fail to compile. So we need to disable the warning.
+EXTRA_CFLAGS += -Wno-incompatible-pointer-types
```

Comment:
Note: I don't think we should ignore the warning. We should fix why it's failing and that's because of a regression in the kernel ethernet device interface. Devome shared a patch updating it [here](https://aur.archlinux.org/packages/r8125-dkms).